cmd/cluster-agent/admissioncontroller/listener_test.go

```diff
@@ -0,0 +1,217 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+package admissioncontroller
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"k8s.io/api/admission/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type responsePayload struct {
+	Response responseObject
+}
+
+type responseObject struct {
+	UID       string
+	Allowed   bool
+	Patch     string
+	PatchType string
+	Status    result
+}
+
+type jsonPath struct {
+	Op    string
+	Path  string
+	Value interface{}
+}
+
+type result struct {
+	Message string
+}
+
+func marshalPodSpec(spec corev1.PodSpec) []byte {
+	pod := corev1.Pod{Spec: spec}
+	body, _ := json.Marshal(pod)
+	return body
+}
+
+func marshalContainers(containers []corev1.Container) []byte {
+	return marshalPodSpec(corev1.PodSpec{Containers: containers})
+}
+
+func newContainer() []corev1.Container {
+	return []corev1.Container{{}}
+}
+
+func newContainerWithVar(name string, value string) []corev1.Container {
+	return []corev1.Container{{Env: []corev1.EnvVar{{Name: name, Value: value}}}}
+}
+
+func expectedPatchFieldPath(name string, fieldPath string) interface{} {
+	return map[string]interface{}{
+		"name": name,
+		"valueFrom": map[string]interface{}{
+			"fieldRef": map[string]interface{}{
+				"fieldPath": fieldPath,
+			},
+		},
+	}
+}
+
+// Base path creation must have an empty array as value
+func expectedBaseEnvPatchValue() interface{} {
+	return []interface{}{}
+}
+
+type ExpectedPatch struct {
+	path  string
+	value interface{}
+}
+
+func TestServe(t *testing.T) {
+	tests := []struct {
+		description     string
+		requestBody     []byte
+		status          int
+		hasResponseBody bool
+		patches         []ExpectedPatch
+		message         string
+	}{
+		{
+			"with container",
```

Comment:
Names of table tests in Go are always one word. They serve more as identifiers, as opposed to how other languages treat them as "the description of the test". Can you please update them all to be just one word? Examples:

* `"ok"`
* `"init"`
* `"conflict"`
* `"no-conflict"`
* `"empty"`
* `"invalid"`